packages/providers/src/transaction-response/transaction-response.ts

```diff
@@ -133,7 +125,14 @@ export class TransactionResponse {
     });
 
     if (!response.transaction) {
-      await this.sleepBasedOnAttempts(++this.fetchAttempts);
+      for await (const statusChange of this.provider.operations.statusChange({
+        transactionId: this.id,
+      })) {
```

Comment:
This can be done to all occurrences to simplify the reading, reinforce the usage, and add context.

```suggestion
      const subscription this.provider.operations.statusChange({ transactionId: this.id });
      for await (const statusChange of subscription) {
```